### PR TITLE
Fix NextJS Message Renderer Resolve Functions

### PR DESCRIPTION
**Pull request by tylergeorges**
**Please describe the changes this PR makes and why it should be merged:**

Adds resolve functions to message renderer to resolve data such as the current guilds members, all roles for the guild, etc.

Resolves FR-79

### CODE REVIEW

File: src/components/Providers/MessageRenderer.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+/* eslint-disable no-alert */
```
Comment (Yomanz) [inline src/components/Providers/MessageRenderer.tsx:1]: Why do you need this?

File: src/hooks/useMessages.ts [hunk 2]
```diff
@@ -43,6 +45,18 @@ export const useMessages = ({
 
   const isReady = data?.channelV2.id === channel;
 
+  const getMentionedMembers = (msgs: Message[]) => {
+    for (const msg of msgs) {
+      const { mentions } = msg;
+
+      if (!mentions.length) continue;
```
Comment (Yomanz) [inline src/hooks/useMessages.ts:52]: Why not just `!mentions`

File: src/state/stores/guild.ts [hunk 3]
```diff
@@ -66,6 +72,7 @@ const guild: GuildStore = {
   currentChannel: undefined,
   refetchGuild: false,
   roles: undefined,
+  members: undefined,
```
Comment (Yomanz) [inline src/state/stores/guild.ts:75]: Can't this be set to `{}` rather than checking for it in the `addMember` action?

**Review by Yomanz — CHANGES_REQUESTED** [review]

File: src/hooks/useMessages.ts (hunk 2 above)
Comment (advaith1) [reply src/hooks/useMessages.ts:52]: !mentions.length checks if mentions is an empty array

Comment (Yomanz) [reply src/hooks/useMessages.ts:52]: Ah ok

**Comment by mason-rogers** [comment]
conflicts here @tylergeorges

**Review by mason-rogers — COMMENTED** [review]
@Yomanz has pointed out my only comment on this already, the rest looks good!

File: src/components/Providers/MessageRenderer.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  const roles = useStoreState(state => state.guild.roles);
+  const guildChannels = useStoreState(state => state.guild.guildChannels);
+
+  const setCurrentThread = useStoreActions(state => state.guild.setCurrentThread);
+  const setIsMembersListOpen = useStoreActions(state => state.ui.setIsMembersListOpen);
+  const setIsDomThreadsPanelOpen = useStoreActions(state => state.ui.setIsDomThreadsPanelOpen);
+  const client = useApolloClient();
+
+  const resolveRole = (id: string): APIRole | null => {
+    if (!roles) return null;
```
Comment (Yomanz) [inline src/components/Providers/MessageRenderer.tsx:41]: This can be 1 if statement

File: src/components/Providers/MessageRenderer.tsx [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+          name
+        }
+      `
+    });
+
+    const mention = client.readFragment({
+      id: `Mention:${id}`,
+
+      fragment: gql`
+        fragment mention on Mention {
```
Comment (Yomanz) [inline src/components/Providers/MessageRenderer.tsx:77]: Fragment should be capitalized

File: src/hooks/useMessages.ts [hunk 6]
```diff
@@ -79,7 +80,10 @@ export const useMessages = ({ guild, channel, threadId }: UseMessagesProps) => {
     }
   }, [fetchMore, messages]);
 
-  const messageState: MessageState = useMemo(() => {
+  let messageState: MessageState;
```
Comment (Yomanz) [inline src/hooks/useMessages.ts:83]: Why is this a let?

File: src/util/convertToDiscord/convertUserToDiscord.ts [hunk 7]
```diff
@@ -1,16 +1,15 @@
...
+
   return {
-    global_name: 'name' in user ? user.name : user.username,
+    global_name: 'username' in user ? user.username : user.name,
     id: user.id,
-    username: 'name' in user ? user.name : user.username,
+    username: 'username' in user ? user.username : user.name,
     discriminator: '',
-    avatar: 'avatarUrl' in user ? getAvatarId(user.avatarUrl) : ''
+    avatar: ''
```
Comment (Yomanz) [inline src/util/convertToDiscord/convertUserToDiscord.ts:13]: No avatar?

**Review by Yomanz — CHANGES_REQUESTED** [review]

**Review by Yomanz — APPROVED** [review]

**Review by mason-rogers — APPROVED** [review]

**Pull request merged**